### PR TITLE
Code Quality: Remove `improvedAutosaves` feature flag code

### LINKED ISSUE #12307: Code Quality: Remove `improvedAutosaves` feature flag code
## Task Description

Now that this has been enabled by default in the last release we can remove the relevant code bits.

### PR DESCRIPTION
**Pull request by timarney**
## Context

Now that this has been enabled by default in #12272  in the last release we can remove the relevant code bits.

## Summary

Removed `improved_autosaves` feature flag

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Ensure Save and AutoSave functionality still works as expected 

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12307

### CODE REVIEW

File: packages/story-editor/src/components/autoSaveHandler/index.js [hunk 1]
```diff
@@ -27,36 +26,23 @@ import { useConfig, useHistory, useStory } from '../../app';
 import useIsUploadingToStory from '../../utils/useIsUploadingToStory';
 
 function AutoSaveHandler() {
-  const improvedAutosaves = useFeature('improvedAutosaves');
   const { autoSaveInterval } = useConfig();
   const {
     state: { hasNewChanges },
   } = useHistory();
-  const { isDraft, saveStory, autoSave } = useStory(
```
Comment (timarney) [inline packages/story-editor/src/components/autoSaveHandler/index.js:35]: @spacedmonkey @swissspidy @miina 

I got paused on this PR as I'm working on the video segmentation PR.

The last test I did --- this wasn't working as expected 

Thinking it's something with removing the isDraft --- but didn't get a chance to dig in.

Any thoughts ?

Comment (swissspidy) [reply packages/story-editor/src/components/autoSaveHandler/index.js:35]: Can you elaborate on "this wasn't working as expected"?

I do see autosave requests being triggered in the Network tab, for both drafts and published stories. So what isn't working exactly?

Comment (miina) [reply packages/story-editor/src/components/autoSaveHandler/index.js:35]: Do you mean it stopped working after removing `isDraft`?

Comment (timarney) [reply packages/story-editor/src/components/autoSaveHandler/index.js:35]: I'll re-test ... I wasn't seeing the revisions count updating.

Comment (timarney) [reply packages/story-editor/src/components/autoSaveHandler/index.js:35]: 🤔 --- re-tested seems to be working 

Seeing revisions and autosave network call 

Will update this and move to review.

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
Note that the `packages/story-editor/src/components/autoSaveHandler/test/autoSaveHandler.js` tests are failing

**Review by miina — APPROVED** [review]

**Comment by timarney** [comment]
I won't be able to get back to this until tonight on my end ...

If anyone wants to push an update to fix the test in the meantime feel fee --- looks like we need to return autoSave vs saveStory
```
 const { renderAgain, saveStory } = setup({});
    
    ```
    
    I pushed another commit earlier to ensure the current call is a function
    https://github.com/GoogleForCreators/web-stories-wp/pull/12357/commits/8407b2ce93636e8bd71b0a7d2299afac62a0b889

**Pull request merged**